cmd/gitops/upgrade/cmd.go

```diff
@@ -37,17 +43,37 @@ func init() {
 
 func upgradeCmdRunE() func(*cobra.Command, []string) error {
 	return func(cmd *cobra.Command, args []string) error {
+		ctx := context.Background()
 		namespace, err := cmd.Parent().Flags().GetString("namespace")
+
 		if err != nil {
 			return fmt.Errorf("couldn't read namespace flag: %v", err)
 		}
 
 		// FIXME: maybe a better way to do this?
 		upgradeCmdFlags.Namespace = namespace
 
+		log := internal.NewCLILogger(os.Stdout)
+		fluxClient := flux.New(osys.New(), &runner.CLIRunner{})
+		factory := services.NewFactory(fluxClient, log)
+
+		providerClient := internal.NewGitProviderClient(os.Stdout, os.LookupEnv, auth.NewAuthCLIHandler, log)
+
+		gitClient, gitProvider, err := factory.GetGitClients(ctx, providerClient, services.GitConfigParams{
+			URL:       upgradeCmdFlags.RepoURL,
```

Comment:
Nice, I think this is simpler to do it as a flag instead of reading from the gitconfig.
Maybe we rename this flag to `--app-config-url` to be consistent w/ `gitops install --app-config-url`.

And we can remove the code that tries to automatically figure this out (`	repoUrlString, err := gitClient.GetRemoteUrl(".", uv.Remote)`).

I don't think we need `RepoOrgAndName` anymore either so we can get rid of the whole `buildUpgradeConfigs` probably. 🤔

Reply:
Thank you very much Simon 🙏 I will add the suggest changes 👍